homeassistant/components/reolink/views.py

```diff
@@ -0,0 +1,138 @@
+@callback
+def async_generate_playback_proxy_url(
+    config_entry_id: str, channel: int, filename: str, stream_res: str, vod_type: str
+) -> str:
+    """Generate proxy URL for event video."""
+
+    url_format = PlaybackProxyView.url
+    return url_format.format(
+        config_entry_id=config_entry_id,
+        channel=channel,
+        filename=filename.replace("/", "|"),
+        stream_res=stream_res,
+        vod_type=vod_type,
+    )
+
+
+class PlaybackProxyView(HomeAssistantView):
+    """View to proxy playback video from Reolink."""
+
+    requires_auth = True
+    url = "/api/reolink/video/{config_entry_id}/{channel}/{stream_res}/{vod_type}/{filename}"
+    name = "api:reolink_playback"
+
+    def __init__(self, hass: HomeAssistant) -> None:
+        """Initialize a proxy view."""
+        self.hass = hass
+        self.session = async_get_clientsession(
+            hass,
+            verify_ssl=False,
+            ssl_cipher=SSLCipherList.INSECURE,
+        )
+
+    async def get(
+        self,
+        request: web.Request,
+        config_entry_id: str,
+        channel: str,
+        stream_res: str,
+        vod_type: str,
+        filename: str,
+        retry: int = 2,
+    ) -> web.StreamResponse:
+        """Get playback proxy video response."""
+        retry = retry - 1
+
+        filename = filename.replace("|", "/")
+        ch = int(channel)
+        host = get_host(self.hass, config_entry_id)
+
+        try:
+            mime_type, reolink_url = await host.api.get_vod_source(
+                ch, filename, stream_res, VodRequestType(vod_type)
+            )
+        except ReolinkError as err:
+            _LOGGER.warning("Reolink playback proxy error: %s", str(err))
+            return web.Response(body=str(err), status=HTTPStatus.BAD_REQUEST)
+
+        if _LOGGER.isEnabledFor(logging.DEBUG):
+            _LOGGER.debug(
+                "Opening VOD stream from %s: %s",
+                host.api.camera_name(ch),
+                host.api.hide_password(reolink_url),
+            )
+
+        try:
+            reolink_response = await self.session.get(
+                reolink_url,
+                timeout=ClientTimeout(
+                    connect=15, sock_connect=15, sock_read=5, total=None
+                ),
+            )
+        except ClientError as err:
+            err_str = host.api.hide_password(
+                f"Reolink playback error while getting mp4: {err!s}"
+            )
+            if retry <= 0:
+                _LOGGER.warning(err_str)
+                return web.Response(body=err_str, status=HTTPStatus.BAD_REQUEST)
+            _LOGGER.debug("%s, renewing token", err_str)
+            await host.api.expire_session(unsubscribe=False)
+            return await self.get(
+                request, config_entry_id, channel, stream_res, vod_type, filename, retry
+            )
+
+        if reolink_response.content_type not in [
+            "video/mp4",
+            "apolication/octet-stream",
```

Comment:
```suggestion
            "application/octet-stream",
```

Has aiohttp constants for them?

Reply:
Actually Reolink is the one who made the spelling error, not me.
The Reolink Home Hub actually sends "apolication" not "application".
This is also in the upstream library: https://github.com/starkillerOG/reolink_aio/blob/febac356797c0bfdf65dae8d08ce3a9351c0fb84/reolink_aio/api.py#L5556C17-L5557

Reply:
I added "application/octet-stream" to the list, in case Reolink fixes the typo in a firmware update.
I also added a comment explaining this is a typo by Reolink and the device actually sends this.